addons/udes_sale_stock/tests/test_manual_sale_order_line_cancellation.py

```diff
@@ -0,0 +1,158 @@
+"""Tests for manual sale order line cancellation"""
+from odoo.exceptions import ValidationError
+
+from . import common
+
+
+class TestManualSaleOrderLineCancellation(common.BaseSaleUDES):
+    """Tests for manual sale order line cancellation"""
+
+    @classmethod
+    def setUpClass(cls):
+        super(TestManualSaleOrderLineCancellation, cls).setUpClass()
+
+        cls.apple_quant = cls.create_quant(
+            cls.apple.id, cls.test_location_01.id, 30, package_id=cls.create_package().id,
+        )
+        cls.banana_quant = cls.create_quant(
+            cls.banana.id, cls.test_location_02.id, 30, package_id=cls.create_package().id,
+        )
+
+        # Create sale order
+        sale = cls.create_sale(cls.customer)
+        cls.apple_sale_line = cls.create_sale_line(sale, cls.apple, 15, route_id=cls.route_out.id)
+        cls.banana_sale_line = cls.create_sale_line(sale, cls.banana, 15, route_id=cls.route_out.id)
+        sale.action_confirm()
+        cls.sale = sale
+        cls.sale.warehouse_id.u_allow_manual_sale_order_line_cancellation = True
+
+        cls.first_picking = sale.picking_ids.filtered(lambda x: x.state == "assigned")
```

Comment:
Super nit: Could we use `p` instead of `x` as we're dealing with pickings?

Reply:
fixed